### ISSUE TITLE
Empty complex elements of sequences with minOccurs not set to 0 are still valid

### ISSUE THREAD

**Issue by ruffsl** [brunato/xmlschema#29]
Complex elements composed of [`xs:sequences`](https://www.w3schools.com/xml/el_sequence.asp) that are empty in the xml document are still considered valid even when the `minOccurs` attribute is not set to 0.

A minimal example:

xsd_string =
``` xml
<?xml version="1.0" encoding="UTF-8"?>
<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema">

    <xs:element name="root" type="Root" />

    <xs:complexType name="Root">
        <xs:sequence>
            <xs:element name="foo" type="xs:string" />
            <xs:element name="bar" type="xs:string" />
        </xs:sequence>
    </xs:complexType>

</xs:schema>
``` 

xml_string_1 = 
```xml
<?xml version="1.0" encoding="UTF-8"?>
<root>
    <foo>my_foo</foo>
    <bar>my_bar</bar>
</root>
```

xml_string_2 = 
``` xml
<?xml version="1.0" encoding="UTF-8"?>
<root>
</root>
```

xml_string_3 = 
``` xml
<?xml version="1.0" encoding="UTF-8"?>
<root>
    <foo>my_foo</foo>
</root>
```

``` python
>>> import xmlschema
>>> print(xmlschema.__version__)
0.9.14
>>> xsd_schema = xmlschema.XMLSchema(xsd_string)
>>> print(xsd_schema.validate(data=xml_string_1, use_defaults=False))
None
>>> print(xsd_schema.validate(data=xml_string_2, use_defaults=False))
None
>>> print(xsd_schema.validate(data=xml_string_3, use_defaults=False))
---------------------------------------------------------------------------
XMLSchemaValidationError                  Traceback (most recent call last)
<ipython-input-39-cc56bcfbbb75> in <module>()
----> 1 print(xsd_schema.validate(data=xml_string_3, use_defaults=False))

/usr/local/lib/python3.5/dist-packages/xmlschema/xsdbase.py in validate(self, data, use_defaults)
    248         """
    249         for error in self.iter_errors(data, use_defaults=use_defaults):
--> 250             raise error
    251 
    252     def iter_errors(self, data, path=None, use_defaults=True):

XMLSchemaValidationError: failed validating <Element 'root' at 0x7f00c75b8b38> with <XsdElement 'bar' at 0x7f00c75bb5c0>.

Reason: tag 'bar' expected.

Schema:

    <xs:element xmlns:xs="http://www.w3.org/2001/XMLSchema" name="bar" type="xs:string" />

Instance:

  <root>
    <foo>my_foo</foo>
  </root>
```

Note that although `xml_string_2` is invalid with respect to the defined schema, it still passes validation when we would have expected to see an error along the lines like:  
```
cvc-complex-type.2.4.b: The content of element 'root' is not complete. One of '{"":foo}' is expected.
```

**Comment by ruffsl** [comment]
I suspect this has something to do with:

https://github.com/brunato/xmlschema/blob/4a8b1b7d82123dcd894856273bcd24ba33618e79/xmlschema/components/groups.py#L285-L290

Given that the while loop may never iterate if the element is empty to begin with.

**Comment by brunato** [comment]
Yes it's the part that you put in your previous comment. Not the simplest thing to fix, because something special have to be done when a schema validation against the meta-schema is performed. For this issue should be enough the following patch (included in the version 0.9.15):

https://github.com/brunato/xmlschema/blob/094caa146ef9a4696eb3e81da0e874b7b3f5eab6/xmlschema/validators/groups.py#L316-L328

For covering the cases of an XSD validation against the meta-schema a more deep analysis will be necessary (type, ref ...). Anyway the schemas are already checked by XSD component classes.

Thanks
